src/gui/sdlmain.cpp

```diff
@@ -1856,10 +1865,17 @@ Bitu GFX_SetSize(int width,
 		assert(emptytex);
 
 		memset(emptytex, 0, texture_area_bytes);
-		glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA8, texsize_w, texsize_h,
+
+		// Using GL_SRGB8_ALPHA8 because GL_SRGB8 doesn't work properly with Mesa drivers on certain integrated Intel GPUs
+		const auto texformat = is_sharp_shader() ? GL_SRGB8_ALPHA8 : GL_RGB8;
+		glTexImage2D(GL_TEXTURE_2D, 0, texformat, texsize_w, texsize_h,
 		             0, GL_BGRA_EXT, GL_UNSIGNED_BYTE, emptytex);
 		delete[] emptytex;
 
+		if (is_sharp_shader()) {
+			glEnable(GL_FRAMEBUFFER_SRGB);
+		}
+
```

Comment:
Suggesting adding a check if SDL_GL_FRAMEBUFFER_SRGB_CAPABLE is active in the context (using `SDL_GL_GetAttribute`), and adding that as a second condition before enabling `GL_FRAMEBUFFER_SRGB`.

Reply:
(Or the alternative if would be conditionally adding the `pow()` gamma correction to the shader, like you suggested in the comments).  

My thoughts are that this situation is probably a very rare corner-case that falling back to the non-corrected-gamma interporation is probably fine.  Worse-case, these users could always switch to pixel-perfect (and no shader) to still get perfectly scaled EGA shading while avoiding any GL shader overhead.

Reply:
I've done the first method as I think conditionally doing the output `pow` is not worth it for such a rare edge case.

Reply:
👍